assets/shaders/octree/coneTracing.glsl

```diff
@@ -168,50 +179,10 @@ void main() {
     outColor = vec4(finalImage.xyz, 1.0);
 }
 
-float SampleShadowMap(sampler2D shadowMap, vec2 coords, float currentDepth, float bias)
-{
-    float pcfDepth = texture(shadowMap, coords).r;
-    float shadow = (currentDepth - bias) > pcfDepth ? 1.0 : 0.0;
-    return shadow;
-}
-
-float SampleShadowMapLinear(sampler2D shadowMap, vec2 coords, float currentDepth, vec2 texelSize, float bias)
-{
-	vec2 pixelPos = coords/texelSize + vec2(0.5);
-	vec2 fracPart = fract(pixelPos);
-	vec2 startTexel = (pixelPos - fracPart) * texelSize;
-
-	float blTexel = SampleShadowMap(shadowMap, startTexel, currentDepth, bias);
-	float brTexel = SampleShadowMap(shadowMap, startTexel + vec2(texelSize.x, 0.0), currentDepth, bias);
-	float tlTexel = SampleShadowMap(shadowMap, startTexel + vec2(0.0, texelSize.y), currentDepth, bias);
-	float trTexel = SampleShadowMap(shadowMap, startTexel + texelSize, currentDepth, bias);
-
-	float mixA = mix(blTexel, tlTexel, fracPart.y);
-	float mixB = mix(brTexel, trTexel, fracPart.y);
-
-	return mix(mixA, mixB, fracPart.x);
-}
-
-float visibilityCalculation(vec4 positionInLightSpace, vec3 normal) {
-    vec3 projectedPosition = positionInLightSpace.xyz / positionInLightSpace.w;
-
-    projectedPosition = projectedPosition * 0.5 + 0.5;
-    float closestDepth = texture(shadowMap, projectedPosition.xy).r;
-    float currentDepth = projectedPosition.z;
-    float bias = max(0.01 * (1.0 - dot(normal, lightDirection)), 0.003);
-    float shadow = 0.0;
-    vec2 texelSize = 1.0 / textureSize(shadowMap, 0);
-    for (int x = -2; x <= 2; x++) {
-        for (int y = -2; y <= 2; y++) {
-            float partialShadow = SampleShadowMapLinear(shadowMap, projectedPosition.xy + vec2(x, y) * texelSize, currentDepth, texelSize, bias);
-            shadow += partialShadow;
-        }
-    }
-    shadow /= 25.0;
-    if (projectedPosition.z > 1.0) {
-        shadow = 0.0;
-    }
-    return 1.0 - shadow;
+float traceShadowCone(vec3 origin, vec3 direction, float targetDistance) {
+    // TODO: Possibly add a little bit in the direction of the normal
```

Comment:
We do this inside the coneTrace itself right?

Reply:
We add a little offset in the direction of the cone, which in this case is the direction to the light and not the normal. This could mean moving it in that offset actually gets it more inside the surface instead of outside which is what we want. Using the actual normal would help.